src/python/pants/engine/legacy/parser.py

```diff
@@ -64,7 +65,7 @@ class Registrar(BuildFileTargetFactory):
       def __init__(self, parse_context, type_alias, object_type):
         self._parse_context = parse_context
         self._type_alias = type_alias
-        self._object_type = object_type
+        self._object_type = object_type if inspect.isclass(object_type) else type(object_type)
```

Comment:
Sometimes Registrar was being passed an instance, rather than a type. To see a use case of this, add a debugger line 

```python
if inspect.isclass(object_type):
  import pdb; pdb.set_trace()
```

In that case, this will grab the underlying type.

Reply:
Hm, this is the result of all the test failures. So the above is not the correct solution.

I'll play around with this tomorrow. 

In particular, I'm curious why in Python 2 the error about `issubclass()` arg 1 having to a class is not triggering, whereas it does in Python 3. When using the repl, `issubclass(1, int)` fails in both Py2 and Py3, so I would think it would fail here too. Yet the error would only raise in Python 3 here.